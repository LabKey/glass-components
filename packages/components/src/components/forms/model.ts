/*
 * Copyright (c) 2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { List, Map, Record } from 'immutable';
import { Option } from 'react-select';
import { Filter } from '@labkey/api';

import { ISelectRowsResult } from '../../query/api';

import { QueryInfo } from '../base/models/QueryInfo';

import { SchemaQuery } from '../base/models/model';

import { DELIMITER } from './input/SelectInput';
import * as actions from './actions';

export interface QuerySelectModelProps {
    allResults: Map<string, Map<string, any>>;
    containerPath?: string;
    displayColumn: string;
    delimiter: string;
    id: string;
    isInit: boolean;
    maxRows: number;
    multiple: boolean;
    preLoad: boolean;
    queryFilters: List<Filter.IFilter>;
    queryInfo: QueryInfo;
    rawSelectedValue: any;
    schemaQuery: SchemaQuery;
    searchResults: Map<string, Map<string, any>>;
    selectedQuery: string;
    selectedItems: Map<string, any>;
    valueColumn: string;
}

export class QuerySelectModel
    extends Record({
        addExactFilter: true,
        allResults: Map<string, Map<string, any>>(),
        containerPath: undefined,
        displayColumn: undefined,
        delimiter: DELIMITER,
        id: undefined,
        isInit: false,
        maxRows: 20,
        multiple: false,
        preLoad: false,
        queryFilters: undefined,
        queryInfo: undefined,
        rawSelectedValue: undefined,
        schemaQuery: undefined,
        searchResults: Map<string, Map<string, any>>(),
        selectedQuery: '',
        selectedItems: Map<string, any>(),
        valueColumn: undefined,
    })
    implements QuerySelectModelProps {
    addExactFilter: boolean;
    allResults: Map<string, Map<string, any>>;
    containerPath: string;
    displayColumn: string;
    delimiter: string;
    id: string;
    isInit: boolean;
    maxRows: number;
    multiple: boolean;
    preLoad: boolean;
    queryFilters: List<Filter.IFilter>;
    queryInfo: QueryInfo;
    rawSelectedValue: any;
    schemaQuery: SchemaQuery;
    searchResults: Map<string, Map<string, any>>;
    selectedQuery: string;
    selectedItems: Map<string, any>;
    valueColumn: string;

    constructor(values?: Partial<QuerySelectModelProps>) {
        super(values);
    }

    formatSavedResults(data?: Map<string, Map<string, any>>, token?: string): Option[] {
        return actions.formatSavedResults(this, data, token);
    }

    getSelectedOptions(): Option | Option[] {
        const options = actions.formatResults(this, this.selectedItems);

        if (this.multiple) {
            return options;
        } else if (options.length === 1) {
            return options[0];
        } else if (options.length > 1) {
            console.warn(
                'QuerySelect.getSelectedOptions: There are more than one options available, however, must in in "multiple" mode.'
            );
        }

        return undefined;
    }

    parseSearch(input: any): boolean | string {
        return actions.selectShouldSearch(this, input);
    }

    saveSearchResults(data: Map<string, Map<string, any>>) {
        return actions.saveSearchResults(this, data);
    }

    setSelection(value: any) {
        return actions.setSelection(this, value);
    }

    search(input: any): Promise<ISelectRowsResult> {
        return actions.fetchSearchResults(this, input);
    }
}

export interface ISelectInitData {
    name: string;
    type: string;
    getSelectComponentId(): string;
}

export interface IUser {
    displayName: string;
    email: string;
    userId: number;
}
