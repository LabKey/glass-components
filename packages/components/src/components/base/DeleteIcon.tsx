import React from 'react';
import { faTimesCircle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface Props {
    id?: string;
    iconCls?: string;
    onDelete: (event) => void;
    title?: string;
}

export class DeleteIcon extends React.Component<Props, any> {
    static defaultProperties = {
        iconCls: 'field-delete',
        title: 'Delete this item',
    };

    render() {
        const { id, title, onDelete, iconCls } = this.props;
        return (
            <span id={id} title={title} className="field-icon" onClick={onDelete}>
                <FontAwesomeIcon className={iconCls} icon={faTimesCircle} />
            </span>
        );
    }
}
