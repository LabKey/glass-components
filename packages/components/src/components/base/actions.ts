/*
 * Copyright (c) 2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { List } from 'immutable';
import { Ajax, Assay, AssayDOM, Utils } from '@labkey/api';

import { AssayDefinitionModel, InferDomainResponse } from './models/model';
import { buildURL } from '../../url/ActionURL';

export function fetchAllAssays(type?: string): Promise<List<AssayDefinitionModel>> {
    return new Promise((res, rej) => {
        Assay.getAll({
            parameters: {
                type
            },
            success: (rawModels: Array<any>) => {
                let models = List<AssayDefinitionModel>().asMutable();
                rawModels.forEach(rawModel => {
                    models.push(AssayDefinitionModel.create(rawModel));
                });
                res(models.asImmutable());
            },
            failure: (error) => {
                rej(error);
            }
        });
    })
}

export function importGeneralAssayRun(assayId: number, file: File, name?: string, comment?: string): Promise<any> {
    return new Promise((resolve, reject) => {
        AssayDOM.importRun({
            assayId,
            name,
            comment,
            files: [file],
            success: (response) => {
                resolve(response)
            },
            failure: (error) => {
                reject(error.exception);
            }
        });
    });
}

export function inferDomainFromFile(file: File, numLinesToInclude: number) : Promise<InferDomainResponse> {
    return new Promise((resolve, reject) => {
        let form = new FormData();
        form.append('file', file);
        form.append('numLinesToInclude', numLinesToInclude ? (numLinesToInclude + 1).toString() : undefined);

        Ajax.request({
            url: buildURL('property', 'inferDomain'),
            method: 'POST',
            form,
            success: Utils.getCallbackWrapper((response) => {
                resolve(InferDomainResponse.create(response));
            }),
            failure: Utils.getCallbackWrapper((error) => {
                console.error(error);
                reject("There was a problem determining the fields in the uploaded file.  Please check the format of the file.");
            })
        });
    })
}

/**
 * This is used for retrieving preview data for a file already on the server side
 * @param file  This can be a rowId for the file, or a path to the file
 * @param numLinesToInclude: the number of lines of data to include (excludes the header)
 */
export function getServerFilePreview(file: string, numLinesToInclude: number) : Promise<InferDomainResponse>{
    return new Promise((resolve, reject) => {

        Ajax.request({
                url: buildURL('property', 'getFilePreview.api'),
                method: 'GET',
                params: {
                    file,
                    numLinesToInclude: numLinesToInclude ? (numLinesToInclude + 1) : undefined // add one to account for the header
                },
                success: Utils.getCallbackWrapper((response) => {
                    resolve(InferDomainResponse.create(response));
                }),
                failure: Utils.getCallbackWrapper((response) => {
                    reject("There was a problem retrieving the preview data.");
                    console.error(response);
                })
            }
        )
    })
}

export function getUserProperties(userId: number): Promise<any> {
    return new Promise((resolve, reject) => {
        return Ajax.request({
            url: buildURL('user', 'getUserProps.api', {userId}, {
                container: '/' // always use root container for this API call
            }),
            success: Utils.getCallbackWrapper((response) => {
                resolve(response);
            }),
            failure: Utils.getCallbackWrapper((response) => {
                reject(response);
            })
        });
    });
}
