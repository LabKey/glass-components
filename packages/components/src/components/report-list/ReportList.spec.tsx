/*
 * Copyright (c) 2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { createMemoryHistory, Route, Router } from 'react-router';
import { mount } from 'enzyme';
import { ReportItemModal, ReportList, ReportListItem } from './ReportList';
import { flattenBrowseDataTreeResponse } from './model';
import exampleData from '../../test/data/example_browse_data_tree_api.json';
import { LoadingSpinner } from '../base/LoadingSpinner';
import { AppURL } from '../../url/AppURL';

const history = createMemoryHistory();

const noop = () => {};
const messageSelector = '.report-list__message';
const createdBySelector = '.report-list-item__person';
const urlMapper = (report) => {
    const { schemaName, queryName, viewName } = report;

    if (!queryName) {
        return null;
    }

    const parts = ['q', schemaName, queryName];

    if (viewName) {
        parts.push(viewName);
    }

    return AppURL.create(...parts);
};

describe('<ReportList />', () => {
    test('flattenBrowseDataTreeResponse works with valid data', () => {
        flattenBrowseDataTreeResponse(exampleData, urlMapper);
    });

    test('Render with no data', () => {
        const component = <ReportList loading={false} reports={[]} onReportClicked={noop}/>;
        const tree = renderer.create(component);
        expect(tree).toMatchSnapshot();
        const wrapper = mount(component);
        expect(wrapper.find(LoadingSpinner)).toHaveLength(0);
        expect(wrapper.find(messageSelector).text()).toContain('No reports');
    });

    test('Render loading', () => {
        const component = <ReportList loading={true} reports={[]} onReportClicked={noop}/>;
        const tree = renderer.create(component);
        expect(tree).toMatchSnapshot();
        const wrapper = mount(component);
        expect(wrapper.find(LoadingSpinner)).toHaveLength(1);
    });

    test('Render with data', () => {
        const reports = flattenBrowseDataTreeResponse(exampleData, urlMapper);
        const component = <ReportList loading={false} reports={reports} onReportClicked={noop}/>;
        const tree = renderer.create(component);
        expect(tree).toMatchSnapshot();
        const wrapper = mount(component);
        expect(wrapper.find(LoadingSpinner)).toHaveLength(0);
        expect(wrapper.find(ReportListItem)).toHaveLength(reports.length);
    });

    test('onReportClicked should execute on click', () => {
        const reports = flattenBrowseDataTreeResponse(exampleData, urlMapper).slice(0, 1);
        const onReportClicked = jest.fn();
        const component = <ReportList loading={false} reports={reports} onReportClicked={onReportClicked}/>;
        const wrapper = mount(component);
        wrapper.find(ReportListItem).simulate('click');
        expect(onReportClicked).toHaveBeenCalledTimes(1);
    });
});

describe('<ReportListItem />', () => {
    test('ReportListItem renders', () => {
        const report = flattenBrowseDataTreeResponse(exampleData, urlMapper)[0];
        const onClick = jest.fn();
        const component = (
            <Router history={history}>
                <Route path="/"  component={() => (<ReportListItem report={report} onClick={onClick}/>)}/>
            </Router>
        );
        const tree = renderer.create(component);
        expect(tree).toMatchSnapshot();
        const wrapper = mount(component);
        expect(wrapper.find(createdBySelector)).toHaveLength(1);
        expect(wrapper.text()).toContain(report.createdBy);
        expect(wrapper.text()).toContain(report.name);
        // Enzyme prefixes relative URLs with http://localhost
        const expectedHref = `http://localhost${report.appUrl.toString()}`;
        expect(wrapper.find('a').getDOMNode()).toHaveProperty('href', expectedHref);
    });

    test('ReportListItem does not render non-existent createdBy', () => {
        const reports = flattenBrowseDataTreeResponse(exampleData, urlMapper);
        const report = reports.filter(r => !r.hasOwnProperty('createdBy'))[0];
        const component = <ReportListItem report={report} onClick={noop}/>;
        const wrapper = mount(component);
        expect(wrapper.find(createdBySelector)).toHaveLength(0);
    });

    test('ReportListItem calls onClick when clicked', () => {
        const report = flattenBrowseDataTreeResponse(exampleData, urlMapper)[0];
        const onClick = jest.fn();
        const component = <ReportListItem report={report} onClick={onClick}/>;
        const wrapper = mount(component);
        wrapper.simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
        // Test that we pass the report to onClick. If this test fails that means we'll need to fix any callbacks that
        // expect a report to be passed to the callback.
        expect(onClick.mock.calls[0][0]).toBe(report);
    });
});

describe('<ReportItemModal />', () => {
    test('ReportItemModal renders', () => {
        const report = flattenBrowseDataTreeResponse(exampleData, urlMapper)[0];
        const component = <ReportItemModal report={report} onClose={noop} />;
        const wrapper = mount(component);
        // Have to use mount + wrapper.debug for snapshot here because react-test-renderer does not work with
        // React portals, which react-bootstrap uses for their modal component.
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
