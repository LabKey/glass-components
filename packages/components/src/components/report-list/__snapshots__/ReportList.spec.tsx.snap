// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportItemModal /> ReportItemModal renders 1`] = `
"<ReportItemModal report={{...}} onClose={[Function: noop]}>
  <div className=\\"report-item-modal\\">
    <Modal show={true} onHide={[Function: noop]} backdrop={true} keyboard={true} autoFocus={true} enforceFocus={true} restoreFocus={true} manager={{...}} renderBackdrop={[Function: renderBackdrop]} animation={true} dialogComponentClass={[Function: ModalDialog]} bsClass=\\"modal\\">
      <Modal onHide={[Function: noop]} keyboard={true} autoFocus={true} enforceFocus={true} restoreFocus={true} manager={{...}} renderBackdrop={[Function: renderBackdrop]} show={true} containerClassName=\\"modal-open\\" transition={[Function: DialogTransition]} backdrop={true} backdropTransition={[Function: BackdropTransition]} backdropClassName=\\"modal-backdrop\\" onEntering={[Function: bound handleEntering]} onExited={[Function: bound handleExited]}>
        <Portal container={[undefined]} onRendered={[Function]}>
          <Portal containerInfo={{...}}>
            <div role=\\"dialog\\" style={[undefined]} className={[undefined]}>
              <BackdropTransition appear={true} in={true}>
                <Fade appear={true} in={true} timeout={150} mountOnEnter={false} unmountOnExit={false}>
                  <Transition appear={true} in={true} timeout={150} mountOnEnter={false} unmountOnExit={false} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: noop]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]}>
                    <div style={[undefined]} className=\\"fade modal-backdrop in\\" onClick={[Function]} />
                  </Transition>
                </Fade>
              </BackdropTransition>
              <RefHolder>
                <DialogTransition appear={true} unmountOnExit={true} in={true} onExit={[undefined]} onExiting={[undefined]} onExited={[Function]} onEnter={[undefined]} onEntering={[Function: bound handleEntering]} onEntered={[undefined]}>
                  <Fade appear={true} unmountOnExit={true} in={true} onExit={[undefined]} onExiting={[undefined]} onExited={[Function]} onEnter={[undefined]} onEntering={[Function: bound handleEntering]} onEntered={[undefined]} timeout={300} mountOnEnter={false}>
                    <Transition appear={true} unmountOnExit={true} in={true} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function]} onEnter={[Function: noop]} onEntering={[Function: bound handleEntering]} onEntered={[Function: noop]} timeout={300} mountOnEnter={false} enter={true} exit={true}>
                      <ModalDialog bsClass=\\"modal\\" style={{...}} className=\\"fade in\\" onClick={[Function: bound handleDialogClick]} role=\\"document\\" tabIndex=\\"-1\\">
                        <div onClick={[Function: bound handleDialogClick]} role=\\"dialog\\" tabIndex=\\"-1\\" style={{...}} className=\\"fade in modal\\">
                          <div className=\\"modal-dialog\\">
                            <div className=\\"modal-content\\" role=\\"document\\">
                              <ModalHeader closeButton={true} closeLabel=\\"Close\\" bsClass=\\"modal-header\\">
                                <div className=\\"modal-header\\">
                                  <CloseButton label=\\"Close\\" onClick={[Function: noop]}>
                                    <button type=\\"button\\" className=\\"close\\" onClick={[Function: noop]}>
                                      <span aria-hidden=\\"true\\">
                                        Ã—
                                      </span>
                                      <span className=\\"sr-only\\">
                                        Close
                                      </span>
                                    </button>
                                  </CloseButton>
                                  <ModalTitle componentClass=\\"h4\\" bsClass=\\"modal-title\\">
                                    <h4 className=\\"modal-title\\">
                                      Example SCR
                                    </h4>
                                  </ModalTitle>
                                </div>
                              </ModalHeader>
                              <UnsupportedReportBody report={{...}}>
                                <ModalBody componentClass=\\"div\\" bsClass=\\"modal-body\\">
                                  <div className=\\"modal-body\\">
                                    <div className=\\"report-list__unsupported-preview\\">
                                      <div className=\\"alert alert-warning unsupported-alert\\">
                                        <div className=\\"unsupported-alert__icon\\">
                                          <span className=\\"fa fa-exclamation-circle\\" />
                                        </div>
                                        <p className=\\"unsupported-alert__message\\">
                                          This report is not currently supported. It is recommended that you view the report in LabKey Server.
                                        </p>
                                        <div className=\\"unsupported-alert__view-link\\">
                                          <a href=\\"/labkey/Biologics/assayreport-view.view?id=37\\" className=\\"btn btn-warning\\">
                                            View in LabKey
                                          </a>
                                        </div>
                                      </div>
                                      <ReportMetadata report={{...}}>
                                        <div className=\\"report-item__metadata\\">
                                          <div className=\\"report-item__metadata-item\\">
                                            <label>
                                              Created By:
                                            </label>
                                            <span>
                                              alanv
                                            </span>
                                          </div>
                                          <div className=\\"report-item__metadata-item\\">
                                            <label>
                                              Type:
                                            </label>
                                            <span>
                                              Sample Comparison
                                            </span>
                                          </div>
                                          <div className=\\"report-item__metadata-item\\">
                                            <label>
                                              Description:
                                            </label>
                                            <span>
                                              From Kevin
                                            </span>
                                          </div>
                                        </div>
                                      </ReportMetadata>
                                    </div>
                                  </div>
                                </ModalBody>
                              </UnsupportedReportBody>
                            </div>
                          </div>
                        </div>
                      </ModalDialog>
                    </Transition>
                  </Fade>
                </DialogTransition>
              </RefHolder>
            </div>
          </Portal>
        </Portal>
      </Modal>
    </Modal>
  </div>
</ReportItemModal>"
`;

exports[`<ReportList /> Render loading 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="report-list"
  >
    <div
      className="report-list__message"
    >
      <span
        className=""
      >
        <i
          aria-hidden="true"
          className="fa fa-spinner fa-spin"
        />
         
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`<ReportList /> Render with data 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="report-list"
  >
    <ul
      className="report-list__list media-list"
    >
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="sample_comparison-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/sample_comparison.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              href="/labkey/Biologics/assayreport-view.view?id=37"
              onClick={[Function]}
            >
              Example SCR
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Injection Volume Comparison 2
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Avg Mass vs pl
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="box_plot-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/box_plot.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Arg Box Plot
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="bar_chart-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/bar_chart.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Bar plot of Mean Cys for Runs
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="pie_chart-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/pie_chart.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Pie chart example
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="bar_chart-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/bar_chart.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Sum of Molecular Weight by Chemical Formula
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_line-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_line.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Line plot: Pro by Sample Date
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="box_plot-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/box_plot.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              MAb Box Plot
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="pie_chart-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/pie_chart.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Distribution by Lot
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Scatter: Arg vs Ala (filtered)
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Scatter: Arg vs Ala
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="xy_scatter-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/xy_scatter.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Injection Volume Comparison
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="box_plot-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/box_plot.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Private Chart
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Test Custom View
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Test Custom View
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Name only
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Private view
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Scientific Name only
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Name only
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              SampleID only
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
      <li
        className="report-list-item media"
        onClick={[Function]}
      >
        <div
          className="media-left"
        >
          <img
            alt="custom_grid-icon"
            className="report-list-item__icon"
            height={null}
            src="/labkey/_images/custom_grid.svg"
            width="100%"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="report-list-item__name media-heading"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Test Custom View
            </a>
          </h4>
          <p
            className="report-list-item__person"
          >
            Created by: 
            alanv
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<ReportList /> Render with no data 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="report-list"
  >
    <div
      className="report-list__message"
    >
      No reports.
    </div>
  </div>
</div>
`;

exports[`<ReportListItem /> ReportListItem renders 1`] = `
<li
  className="report-list-item media"
  onClick={[Function]}
>
  <div
    className="media-left"
  >
    <img
      alt="xy_scatter-icon"
      className="report-list-item__icon"
      height={null}
      src="/labkey/_images/xy_scatter.svg"
      width="100%"
    />
  </div>
  <div
    className="media-body"
  >
    <h4
      className="report-list-item__name media-heading"
    >
      <a
        href="/q/assay.general.sec/data"
        onClick={[Function]}
        style={Object {}}
      >
        Injection Volume Comparison 2
      </a>
    </h4>
    <p
      className="report-list-item__person"
    >
      Created by: 
      alanv
    </p>
  </div>
</li>
`;
